core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -226,9 +226,9 @@ private[spark] class BlockManager(
   private val maxFailuresBeforeLocationRefresh =
     conf.get(config.BLOCK_FAILURES_BEFORE_LOCATION_REFRESH)
 
-  private val slaveEndpoint = rpcEnv.setupEndpoint(
+  private val replicaEndpoint = rpcEnv.setupEndpoint(
```

Comment:
You've probably thought this through more than I, but is 'replica' the terminology used elsewhere in this component? I see replacements with 'agent' and 'worker' elsewhere, which are fine per se, just making sure they are differing in order to match, where applicable, terminology used that way in the relevant component.

Reply:
We have `spark.storage.replication.*` which controls how blocks are replicated and similar things like `replicateBlock`, but I think replica it's self is good. We could also go with "shard" here if we wanted?

Reply:
This only a common BlockManager at executor side. Why it has to be specific to 'replica'?  Besides, replication also is disabled by default.  Therefore, it also doesn't make sense to me to change the name form 'BlockManagerSlaveEndpoint' to 'BlockManagerReplicaEndpoint' if we're searching a synonyms for 'Slave'.

Reply:
Yeah the goal was primary/replica. We could go with “Storage” instead if you think that would be better? Since those endpoints are responsible for storing the blocks.

Reply:
I like "storageEndpoint". I think that's probably more accurate than replica since we don't necessarily replicate blocks, and this is the endpoint for the primary as well.

Reply:
Sounds good. I’ll leave this until early next week incase anyone has additional thoughts here and then I’ll update the names :)

Reply:
+1 for `storageEndpoint`

Reply:
+1 for storageEndpoint